driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedTestFailureValidator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.unified;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.lang.Nullable;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Collection;
+
+public class UnifiedTestFailureValidator extends UnifiedTest {
+    private final String fileDescription;
+    private final String testDescription;
+    private Throwable setupException;
+
+    public UnifiedTestFailureValidator(final String fileDescription, final String testDescription, final String schemaVersion,
+                                       @Nullable final BsonArray runOnRequirements, final BsonArray entities, final BsonArray initialData,
+                                       final BsonDocument definition) {
+        super(schemaVersion, runOnRequirements, entities, initialData, definition);
+        this.fileDescription = fileDescription;
+        this.testDescription = testDescription;
+    }
+
+    @Before
+    public void setUp() {
+        try {
+            super.setUp();
+        } catch (Throwable e) {
```

Comment:
I'll leave the general discussion of `Throwable` for another time, but  your question made me think about whether we could get rid of Throwable in this particular spot. It's not so straightforward because, for better or worse, AssertionError extends Error and not RuntimeException, and we have to handle both.  But I rewrote the class a bit to make it more explicit.  Let me know if you prefer this approach.

Reply:
Note that the field still needs to be of type `Throwable`.  I don't see a way around that.

Reply:
Thanks!